idx/memory/memory.go

```diff
@@ -318,31 +321,38 @@ func (m *MemoryIdx) UpdateArchive(archive idx.Archive) {
 	*(m.defById[archive.Id]) = archive
 }
 
+// get or add an object in the interning store
+// return a string with data pointed to the interned data
+// this assumes that no compression is used in the store
+func (m *MemoryIdx) internAcquire(sz string) (string, error) {
+	acquired, err := idx.IdxIntern.AddOrGetString([]byte(sz))
+	if err != nil {
+		return sz, err
+	}
+
+	return acquired, nil
+}
+
 // indexTags reads the tags of a given metric definition and creates the
 // corresponding tag index entries to refer to it. It assumes a lock is
 // already held.
-func (m *MemoryIdx) indexTags(def *schema.MetricDefinition) {
+func (m *MemoryIdx) indexTags(def *idx.MetricDefinition) {
 	tags, ok := m.tags[def.OrgId]
 	if !ok {
 		tags = make(TagIndex)
 		m.tags[def.OrgId] = tags
 	}
 
 	for _, tag := range def.Tags {
-		tagSplits := strings.SplitN(tag, "=", 2)
-		if len(tagSplits) < 2 {
-			// should never happen because every tag in the index
-			// must have a valid format
-			invalidTag.Inc()
-			log.Errorf("memory-idx: Tag %q of id %q has an invalid format", tag, def.Id)
-			continue
-		}
-
-		tagName := tagSplits[0]
-		tagValue := tagSplits[1]
+		// we don't care if an error is returned for now
+		// because the original string will be returned
+		// and at least the process can still continue
+		tagName, _ := m.internAcquire(tag.Key)
```

Comment:
I think if there's an error returned here then this is likely indicating a serious issue. If nothing else can be done, I'd at least increase some counter or log an msg.